src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -48,11 +51,19 @@ def __init__(self, post_processor: PostProcessor | None = None) -> None:
         self.loss: nn.Module
         self.callbacks: list[Callback]
 
-        self.image_metrics: AnomalibMetricCollection
-        self.pixel_metrics: AnomalibMetricCollection
-
+        # set the post-processor
         self.post_processor = post_processor or self.default_post_processor()
 
+        # set the evaluator
+        self.evaluator: Evaluator | None
+        if isinstance(evaluator, Evaluator):
+            self.evaluator = evaluator
+        elif isinstance(evaluator, bool):
+            self.evaluator = self.configure_evaluator() if evaluator else None
+        else:
+            msg = f"evaluator must be of type Evaluator or bool, got {type(evaluator)}"
+            raise TypeError(msg)
```

Comment:
As per @ashwinvaidya17's comment on the pre-processor pr, the same goes for this one as well